assets/backport/github/github.go

```diff
@@ -0,0 +1,311 @@
+
+	updatedCommit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		headBranchCommit.GetSHA())
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	// Create the actual cherry-pick commit on the target branch containing the merge commit tree.
+	commit, _, err := c.Client.Git.CreateCommit(ctx, c.Organization, c.Repository, &go_github.Commit{
+		Message: cherryCommit.Message,
+		Tree:    mergeTree,
+		Parents: []*go_github.Commit{
+			updatedCommit,
+		},
+	})
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	// Overwrite the merge commit and its parent on the branch by the newly created commit.
+	// The result will be equivalent to what would have happened with a fast-forward merge.
+	sha := commit.GetSHA()
+	refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+	_, _, err = c.Client.Git.UpdateRef(ctx, c.Organization, c.Repository, &go_github.Reference{
+		Ref: go_github.String(refName),
+		Object: &go_github.GitObject{
+			SHA: go_github.String(sha),
+		},
+	}, true)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	return mergeTree, sha, nil
+}
+
+// createSiblingCommit creates a commit with the passed in commit's tree and parent
+// and updates the passed in branch to point at that commit.
+func (c *Client) createSiblingCommit(ctx context.Context, branchName string, branchHeadCommit *go_github.Commit, cherryParent *go_github.Commit) error {
+	tree := branchHeadCommit.GetTree()
+
+	// This will be the temp commit, commit is lost.
+	commit, _, err := c.Client.Git.CreateCommit(ctx, c.Organization, c.Repository, &go_github.Commit{
+		Message: go_github.String("field-not-required"),
+		Tree:    tree,
+		Parents: []*go_github.Commit{
+			cherryParent,
+		},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	sha := commit.GetSHA()
+
+	refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+	_, _, err = c.Client.Git.UpdateRef(ctx, c.Organization, c.Repository, &go_github.Reference{
+		Ref: go_github.String(refName),
+		Object: &go_github.GitObject{
+			SHA: go_github.String(sha),
+		},
+	}, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// createBranchFrom creates a branch from the passed in branch's HEAD.
+func (c *Client) createBranchFrom(ctx context.Context, branchFromName string, newBranchName string) error {
+	baseBranch, _, err := c.Client.Repositories.GetBranch(ctx, c.Organization, c.Repository, branchFromName, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	newRefBranchName := fmt.Sprintf("%s%s", branchRefPrefix, newBranchName)
+	baseBranchSHA := baseBranch.GetCommit().GetSHA()
+
+	ref := &go_github.Reference{
+		Ref: go_github.String(newRefBranchName),
+		Object: &go_github.GitObject{
+			SHA: go_github.String(baseBranchSHA), /* SHA to branch from */
+		},
+	}
+	_, _, err = c.Client.Git.CreateRef(ctx, c.Organization, c.Repository, ref)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// merge merges a branch.
```

Comment:
```suggestion
// merge merges a branch at `headCommitSHA` into branch `base`.
```
Is that around the right way?

Reply:
Yes.